src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        // required variables - are not class variables, because
+        // they do not reflect the state of the class
+        int supply = 0;
+        int buy = 0;
+
+        String[] lines = readFile(fromFileName).split(System.lineSeparator());
+
+        // calculating
+        for (String stringLine : lines) {
+            String[] parsingLine = stringLine.split(SEPARATOR);
+            int amount = Integer.parseInt(parsingLine[AMOUNT_INDEX]);
+            switch (parsingLine[OPERATION_TYPE_INDEX]) {
+                case SUPPLY:
+                    supply += amount;
+                    break;
+                case BUY:
+                    buy += amount;
+                    break;
+                default:
+                    throw new RuntimeException("This type of operation is not supported");
+            }
+        }
+
+        this.writeFile(toFileName, this.createReport(supply, buy));
```

Comment:
![image](https://github.com/mate-academy/jv-work-with-file/assets/156338484/cabb9037-0320-4c1d-bc0f-b7d9e02c00bd)
Try to use this template for make good structure

Reply:
parsing and calculation you can do in create report or add one more private method is it needed, up2u